openhands/controller/stuck.py

```diff
@@ -12,6 +12,8 @@
 from openhands.events.observation.error import ErrorObservation
 from openhands.events.observation.observation import Observation
 
+MAX_STUCK_ITERATIONS = 4
```

Comment:
It seems this PR isn't changing some checks for 3 actions/obs. So adding this here is a bit misleading, it feels like it will check for 4 all the time or almost (except the scenario with 6) but it doesn't; there still are sometimes checks for 3, sometimes checks for 4.
IMHO we shouldn't do it, or add a comment and perhaps some better name, dunno, maybe even FOUR_ITERATIONS so we know what its role is.

Reply:
Where does it stop on 3? 🤔

Reply:
`_is_stuck_monologue`

Reply:
Ok, moved the call of `_is_stuck_monologue() ` up, so that it can fire on 3 again.